script/cibuild

```diff
@@ -74,37 +98,68 @@ main ()
 
         while read scheme
         do
-            build_scheme "$scheme" || status=1
+            # Build each scheme, but stop the entire build if one returns
+            # a nonzero status code
+            build_scheme "$scheme" || status=1 || break
         done
 
         exit $status
     )
 }
 
+#
+# Outputs the first file or directory in pwd that matches the given argument.
+#
 find_pattern ()
 {
     ls -d $1 2>/dev/null | head -n 1
 }
 
-run_xctool ()
+#
+# Invokes xcodebuild using either XCWORKSPACE or XCODEPROJ passing BUILD_OPTIONS
+# and relaying any arguments as additional command-line parameters.
+#
+run_build_tool ()
 {
+    local XCPRETTY_FORMAT_OPTIONS=
+
+    if [ -n "$TRAVIS" ]
+    then
+        # Use a special formatter when running on TravisCI
+        gem install xcpretty-travis-formatter
```

Comment:
It looks like this install command is run every time `run_build_tool` is called. It might be good to add a check like for vanilla `xcpretty` above. Another option would be to add this to the `.travis.yml` file since this would only be run on Travis anyway.